tensorboard/plugins/text/summary_test.py

```diff
@@ -0,0 +1,144 @@
+        name, data, display_name=display_name, description=description))
+    pb_via_op = self.normalize_summary_pb(
+        self.pb_via_op(op, feed_dict=feed_dict))
+    self.assertProtoEquals(pb, pb_via_op)
+    return pb
+
+  def test_metadata(self):
+    pb = self.compute_and_check_summary_pb('do', 'A deer. A female deer.')
+    summary_metadata = pb.value[0].metadata
+    plugin_data = summary_metadata.plugin_data
+    self.assertEqual(summary_metadata.display_name, 'do')
+    self.assertEqual(summary_metadata.summary_description, '')
+    self.assertEqual(plugin_data.plugin_name, metadata.PLUGIN_NAME)
+    content = summary_metadata.plugin_data.content
+    # There's no content, so successfully parsing is fine.
+    metadata.parse_plugin_metadata(content)
+
+  def test_explicit_display_name_and_description(self):
+    display_name = '"Re"'
+    description = 'A whole step above do.'
+    pb = self.compute_and_check_summary_pb('re', 'A drop of golden sun.',
+                                           display_name=display_name,
+                                           description=description)
+    summary_metadata = pb.value[0].metadata
+    self.assertEqual(summary_metadata.display_name, display_name)
+    self.assertEqual(summary_metadata.summary_description, description)
+    plugin_data = summary_metadata.plugin_data
+    self.assertEqual(plugin_data.plugin_name, metadata.PLUGIN_NAME)
+    content = summary_metadata.plugin_data.content
+    # There's no content, so successfully parsing is fine.
+    metadata.parse_plugin_metadata(content)
+
+  def test_string_value(self):
+    pb = self.compute_and_check_summary_pb('mi', 'A name I call myself.')
+    value = tf.make_ndarray(pb.value[0].tensor).item()
+    self.assertIsInstance(value, six.binary_type)
+    self.assertEqual(b'A name I call myself.', value)
+
+  def test_np_array_string_value(self):
+    pb = self.compute_and_check_summary_pb('fa', 'A long, long way to run.')
+    value = tf.make_ndarray(pb.value[0].tensor).item()
+    self.assertIsInstance(value, six.binary_type)
+    self.assertEqual(b'A long, long way to run.', value)
+
+  def test_np_array_unicode_value(self):
+    pb = self.compute_and_check_summary_pb('fa', u'A long, long way to run.')
+    value = tf.make_ndarray(pb.value[0].tensor).item()
+    self.assertIsInstance(value, six.binary_type)
+    self.assertEqual(b'A long, long way to run.', value)
+
+  def test_non_string_value_in_op(self):
+    with six.assertRaisesRegex(
+        self,
+        Exception,
+        r'Const:0 must be of type <dtype: \'string\'>'):
+      with tf.Session() as sess:
+        sess.run(summary.op('so', tf.constant(5)))
+
+  def test_non_string_value_in_pb(self):
+    with six.assertRaisesRegex(
+        self,
+        ValueError,
+        r'Type \'int\d+\' is not supported. Only strings are.'):
+      summary.pb('la', np.array(range(42)))
+
+  def test_unicode_numpy_array_value_in_pb(self):
```

Comment:
What is this for? Isn't it redundant with `test_np_array_unicode_value`? (Also, there aren't any numpy arrays here, either.)

Reply:
Ah, indeed. That test is misleading and unnecessary. Done.